### PR TITLE
Docker refactoring, Improving Ubuntu image, add better docker testing, etc ..

### PR DESCRIPTION
**Pull request by ImreSamu**
Ready to review / refactoring / etc ... 

#### Main changes:
- Update Alpine to version 3.19.
- Refactor DOCKER_PG_LLVM_DEPS based on upstream PostgreSQL fixes.
- Enhancements to both Dockerfile and Dockerfile.ubuntu.
- Resolve Dockerfile.ubuntu issues: 
   * ubuntu:devel UTF-8 error via initdb ; so refactored this part of code.
   * calling gosu  `RUN sed -i -e 's/su-exec/gosu/g' "/usr/local/bin/docker-entrypoint.sh"`
- Reduce ubuntu /alpine -  docker image sizes.
- Integrate Docker Official Postgres testing (./test/*)
  - see   `./test/docker_test.md`
- Improve `ci/local_docker_matrix.sh` with enhanced logging and testing capabilities.
- Improve .dockerignore ,  .gitignore

#### Todo later:
* Adding "Docker Official Postgres testing (./test/*) " to the CI
    -   .github/workflows/docker.yml
    -   .github/workflows/dockertest.yml

#### Not tested changes :

Compared to changes in other code, I have not tested this changes at all: 
*  .github/workflows/docker.yml   < ----  Need extra attention! 

    
####  File changes: 
````   
modified:   .dockerignore
modified:   .github/workflows/docker.yml
modified:   .github/workflows/dockertest.yml
modified:   .gitignore
modified:   Dockerfile
modified:   Dockerfile.ubuntu
modified:   ci/local_docker_matrix.sh
new file:   test/docker_test.md
new file:   test/orioledb-config.sh
new file:   test/tests/orioledb-basics/run.sh
````

####  ci/local_docker_matrix.sh results .

Test matrix result, which I tested locally - with the created Docker images:

```
orioletest   15-clang-alpine-3.13    701df24531f4   55 minutes ago      205MB
orioletest   15-clang-alpine-3.14    c184466b98de   58 minutes ago      233MB
orioletest   15-clang-alpine-3.15    8104542eff01   About an hour ago   243MB
orioletest   15-clang-alpine-3.16    f3d44ca608ac   About an hour ago   249MB
orioletest   15-clang-alpine-3.17    efb65b19cbb7   About an hour ago   273MB
orioletest   15-clang-alpine-3.18    bc9b655ac0af   About an hour ago   280MB
orioletest   15-clang-alpine-3.19    d6ac1f8bbfa1   About an hour ago   291MB
orioletest   15-clang-ubuntu-20.04   165629a5bdb5   43 minutes ago      744MB
orioletest   15-clang-ubuntu-22.04   090ddfbbfffd   46 minutes ago      541MB
orioletest   15-clang-ubuntu-23.10   b965bce93eaa   50 minutes ago      658MB
orioletest   15-clang-ubuntu-devel   953bef0b6f27   37 minutes ago      678MB
orioletest   15-gcc-alpine-3.13      fe239534834f   14 minutes ago      205MB
orioletest   15-gcc-alpine-3.14      e884d7e20e2f   17 minutes ago      234MB
orioletest   15-gcc-alpine-3.15      400a34175bbf   20 minutes ago      243MB
orioletest   15-gcc-alpine-3.16      02887b39b97a   23 minutes ago      249MB
orioletest   15-gcc-alpine-3.17      af5231d39048   28 minutes ago      273MB
orioletest   15-gcc-alpine-3.18      6f5e0d5f900f   31 minutes ago      281MB
orioletest   15-gcc-alpine-3.19      1e067db460c5   34 minutes ago      292MB
orioletest   15-gcc-ubuntu-20.04     49f7d21e4cab   4 minutes ago       746MB
orioletest   15-gcc-ubuntu-22.04     8baec6c00995   7 minutes ago       542MB
orioletest   15-gcc-ubuntu-23.10     d2d2e736730e   11 minutes ago      659MB
orioletest   15-gcc-ubuntu-devel     4c0ef44dbff6   27 seconds ago      679MB
orioletest   16-clang-alpine-3.13    9696736c5d80   2 hours ago         206MB
orioletest   16-clang-alpine-3.14    46c1d2ecfb80   2 hours ago         235MB
orioletest   16-clang-alpine-3.15    8f2609ee95b9   2 hours ago         245MB
orioletest   16-clang-alpine-3.16    2589190b2b04   3 hours ago         251MB
orioletest   16-clang-alpine-3.17    fde59f2ba981   3 hours ago         275MB
orioletest   16-clang-alpine-3.18    108f02324072   3 hours ago         282MB
orioletest   16-clang-alpine-3.19    b01615fdd2b0   3 hours ago         293MB
orioletest   16-clang-ubuntu-20.04   1a7612e3944d   2 hours ago         746MB
orioletest   16-clang-ubuntu-22.04   ad53b1506ad7   2 hours ago         543MB
orioletest   16-clang-ubuntu-23.10   64026e110f59   2 hours ago         660MB
orioletest   16-clang-ubuntu-devel   5553f3859a70   2 hours ago         680MB
orioletest   16-gcc-alpine-3.13      473d8fdeb639   2 hours ago         207MB
orioletest   16-gcc-alpine-3.14      54c7d8706370   2 hours ago         236MB
orioletest   16-gcc-alpine-3.15      24210ed5838d   2 hours ago         245MB
orioletest   16-gcc-alpine-3.16      0a4a48286e50   2 hours ago         251MB
orioletest   16-gcc-alpine-3.17      6670ea6413c9   2 hours ago         275MB
orioletest   16-gcc-alpine-3.18      2757d45a2cc5   2 hours ago         283MB
orioletest   16-gcc-alpine-3.19      890430cdbcc4   2 hours ago         294MB
orioletest   16-gcc-ubuntu-20.04     b2d1d8bd229d   About an hour ago   748MB
orioletest   16-gcc-ubuntu-22.04     365de143aedd   2 hours ago         544MB
orioletest   16-gcc-ubuntu-23.10     6797afa32753   2 hours ago         661MB
orioletest   16-gcc-ubuntu-devel     91d248187e88   About an hour ago   681MB
REPOSITORY   TAG                     IMAGE ID       CREATED             SIZE

real	172m15.301s
user	1m7.633s
sys	0m56.984s
```

### CODE REVIEW

**Comment by ImreSamu** [comment]
I fixed the .github/workflows/dockertest.yml. 

Now it runs successfully in my CI environment. 
https://github.com/ImreSamu/orioledb/actions/runs/8105134579

File: .dockerignore [hunk 1]
```diff
@@ -46,4 +55,26 @@ output_iso/
 include/utils/stopevents_defs.h
 include/utils/stopevents_data.h
 orioledb.typedefs
-!**/ci/antithesis/libvoidstar.so
\ No newline at end of file
+!**/ci/antithesis/libvoidstar.so
```
Comment (homper) [inline .dockerignore]: Sorry, it was copied from .gitignore not in this request, but it's probably better to replace it with just `ci/antithesis` (This is not needed for docker image files)

**Comment by pashkinelfe** [comment]
I see several tests fail under valgrind. It seems like Postgres is stuck at stopping. I couldn't see this fail on the other branches, though there were some valgrind timeout failures.

**Comment by akorotkov** [comment]
I think this is because this pull request was based on the `main` branch lacking the many bug fixes we have in `pre_merge`.  I've merged `pre_merge` into `main`.

**Comment by pashkinelfe** [comment]
Tests of this PR passed. I think it's ready to be merged.

**Comment by akorotkov** [comment]
Merged in f2820112c7.

**Comment by ImreSamu** [comment]
@akorotkov :
> Merged in f282011.

I believe the f282011 commit 
is from a previous PR made a year ago, committed on June 11, 2023. 
This PR is an update to that from 2024.

**Comment by akorotkov** [comment]
@ImreSamu, sorry, I will recheck!

**Comment by akorotkov** [comment]
Merged in 6544ef4041.
@ImreSamu, thank you so much for your contribution!

**Pull request closed without merging**